sdk/Context.go

```diff
@@ -422,6 +431,70 @@ func (c *Context) GetVariableKeys() (map[string]string, error) {
 	return variableKeys, nil
 }
 
+func removeDuplicateValues(intSlice []string) []string {
+	keys := make(map[string]bool)
+	var list []string
+
+	for _, entry := range intSlice {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			list = append(list, entry)
+		}
+	}
+	return list
+}
+
+func (c *Context) GetCustomFieldValueKeys() ([]string, error) {
+	var err = c.CheckReady(true)
+	if err != nil {
+		return nil, err
+	}
+
+	var keys []string
+
+	for _, experiment := range c.Data_.Experiments {
```

Comment:
Shouldn't this be acquiring the data lock?

Reply:
Done